packages/next-codemod/transforms/next-lint-to-explicit-eslint.ts

```diff
@@ -0,0 +1,197 @@
+  },
+];
+
+export default eslintConfig;
+`
+
+const ESLINT_CONFIG_TEMPLATE_JAVASCRIPT = `import { dirname } from "path";
+import { fileURLToPath } from "url";
+import { FlatCompat } from "@eslint/eslintrc";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+});
+
+const eslintConfig = [
+  ...compat.extends("next/core-web-vitals"),
+  {
+    ignores: [
+      "node_modules/**",
+      ".next/**",
+      "out/**",
+      "build/**",
+      "next-env.d.ts",
+    ],
+  },
+];
+
+export default eslintConfig;
+`
+
+function detectTypeScript(projectRoot: string): boolean {
+  return existsSync(path.join(projectRoot, 'tsconfig.json'))
+}
+
+function hasExistingEslintConfig(projectRoot: string): boolean {
+  const eslintConfigs = [
+    'eslint.config.js',
+    'eslint.config.mjs',
+    'eslint.config.cjs',
+    'eslint.config.ts',
+    'eslint.config.mts',
+    'eslint.config.cts',
+    '.eslintrc.js',
+    '.eslintrc.cjs',
+    '.eslintrc.yaml',
+    '.eslintrc.yml',
+    '.eslintrc.json',
+    '.eslintrc',
+  ]
+
+  return eslintConfigs.some((config) =>
+    existsSync(path.join(projectRoot, config))
+  )
+}
+
+function updatePackageJsonScripts(packageJsonContent: string): {
+  updated: boolean
+  content: string
+} {
+  try {
+    const packageJson = JSON.parse(packageJsonContent)
+    let needsUpdate = false
+
+    if (!packageJson.scripts) {
+      packageJson.scripts = {}
+    }
+
+    // Update lint script from "next lint" to "eslint ."
+    if (packageJson.scripts.lint === 'next lint') {
+      packageJson.scripts.lint = 'eslint .'
+      needsUpdate = true
+    }
+
+    // Ensure required devDependencies exist
+    if (!packageJson.devDependencies) {
+      packageJson.devDependencies = {}
+    }
+
+    if (!packageJson.devDependencies.eslint) {
+      packageJson.devDependencies.eslint = '^9'
+      needsUpdate = true
+    }
+
+    if (!packageJson.devDependencies['eslint-config-next']) {
+      // Use the same version as next if available
+      const nextVersion =
+        packageJson.dependencies?.next || packageJson.devDependencies?.next
+      packageJson.devDependencies['eslint-config-next'] =
+        nextVersion || 'latest'
+      needsUpdate = true
+    }
+
+    if (!packageJson.devDependencies['@eslint/eslintrc']) {
+      packageJson.devDependencies['@eslint/eslintrc'] = '^3'
+      needsUpdate = true
+    }
+
+    const updatedContent = JSON.stringify(packageJson, null, 2) + '\n'
+    return { updated: needsUpdate, content: updatedContent }
+  } catch (error) {
+    console.error('Error updating package.json:', error)
+    return { updated: false, content: packageJsonContent }
+  }
+}
+
+export default function transformer(files: string[], _options: any): void {
+  if (files.length !== 1) {
+    console.error('Error: Please specify exactly one directory path')
+    return
+  }
+
+  const projectRoot = path.resolve(files[0])
```

Comment:
Shouldn't it be `dirname`?

```suggestion
  const projectRoot = path.dirname(files[0])
```

Reply:
I don't think so, but I could be mistaken. Since `files[0]` contains the directory name, then `path.resolve` just returns it, but `path.dirname` would give the parent directory.

```
Input: "/Users/me/code/next.js"
  path.resolve("/Users/me/code/next.js"): /Users/me/code/next.js
  path.dirname("/Users/me/code/next.js"): /Users/me/code
```

Reply:
Got it, thought `files` was an array of file paths. Better to be something like `dirs`? 😅

related: https://github.com/vercel/next.js/pull/82685/files#r2279934814